doc/manual_experimental.md

```diff
@@ -659,6 +659,22 @@ Let `source` be one of:
 - A constant array/seq/object/tuple constructor.
 
 
+When modifying a location of a `string` type via `toOpenArray`,
+`prepareMutation` is needed before modification.
```

Comment:
this is .. _really_ quite undesireable as far as designs go, to the point that the feature is practically unusable in real code. I'll assume it's just a temporary restriction?

Reply:
Those pesky copy on write string literals, might make sense in the future to make a view into a string implicitly call `prepareMutation`. It's a slight overhead but it's much preferable to the alternative in my view.

Reply:
See also discussion on https://github.com/nim-lang/Nim/pull/20621

Reply:
Fine, we can emit the call automatically. But people did complain about that too, just so that you know. :P

Reply:
well, either that, or make strings non-cow - ie string literals are typically small and rarely used in contexts where performance _actually_ matters - of course you can put a string literal copy in a tight loop in a microbenchmark, but that just doesn't happen in real life - the performance tradeoff is really weird, ie penalise all usages of strings (which typically are not literals) so that the rare string literal copy (when used with var!) can be avoided...

Reply:
There are idioms like

```nim

  case x
  of value1: result = "abc"
  ...

```

Which benefit from it. I would rather make strings full COW instead of making them even less COW.

Reply:
I'm sure there are - I still find it hard to believe that you'll find them used in any context where they make a measurable impact on performance that isn't eaten up a hundred times over by the noop cow preparations for _non-literals_ - in particular, if you pass a literal to a `let` or to a non-var openarray, there should still be no copy involved - ditto for "pure" functions, once the compiler becomes clever enough to identify those (basically, `let x = selectLiteral(condition)` shouldn't require a copy either as long as the `selectLiteral` function only returns literals - this can be detected by the compiler and optimized accordingly rather than introducing a massive gotcha for a truly unique combination of inputs.

Reply:
> full COW

full COW at least can lead to benefits in some cases that matter, and give the string type a raison d'etre beyond `seq[char]` - but even then it should be transparent to the user in most cases - ie one shouldn't have to call a special function to repeat that `var` is intended and not just for fun.

Reply:
Yes, so the compiler needs to emit `prepareMutation` on its own. It already does that in many cases, it's just that half-way through I changed my mind based on feedback. But ok.